### PR TITLE
Support Intan "one-file-per-signal" and "one-file-per-channel" formats.

### PR DESCRIPTION
**Pull request by h-mayorquin**
As in the title. We need new releases of spikeinterface and neo for this.

Adding tests here to see if anything else is broken. 

Related to issues:
#789

### CODE REVIEW

**Comment by h-mayorquin** [comment]
With the release of neo this is ready for review.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 # Upcoming
 
+### Features
```
Comment (CodyCBakerPhD) [inline CHANGELOG.md:3]: I prefer having deprecations front and center at the top of the changelog for easier reference

File: requirements-testing.txt [hunk 2]
```diff
@@ -3,5 +3,5 @@ pytest-cov
 ndx-events>=0.2.0  # for special tests to ensure load_namespaces is set to allow NWBFile load at all timess
 parameterized>=0.8.1
 ndx-miniscope
-spikeinterface[qualitymetrics]>=0.100.0
+spikeinterface[qualitymetrics] @ git+https://github.com/SpikeInterface/spikeinterface.git@main
```
Comment (CodyCBakerPhD) [inline requirements-testing.txt:6]: So you're proposing we merge with only the testing requirements pinned to latest SI dev?

File: src/neuroconv/datainterfaces/ecephys/requirements.txt [hunk 3]
```diff
@@ -1 +1,2 @@
-spikeinterface>=0.100.0
+spikeinterface>=0.100.1,<0.100.6
+neo>=0.13.1
```
Comment (CodyCBakerPhD) [inline src/neuroconv/datainterfaces/ecephys/requirements.txt:2]: But then for release purposes won't we _need_ `spikeinterface>=0.100.6`?

**Comment by CodyCBakerPhD** [comment]
I thought this was a multi-file style of the format? So shouldn't there be a new interface for Intan that takes `folder_path` in that case?

File: requirements-testing.txt (hunk 2 above)
Comment (h-mayorquin) [reply requirements-testing.txt:6]: I am not sure why I did this or whether it is necessary. Removing it.

File: CHANGELOG.md (hunk 1 above)
Comment (h-mayorquin) [reply CHANGELOG.md:3]: I was not aware on what you preferred. You want this after deprecations? Your way seems to be numpy way btw:

https://numpy.org/doc/stable/release.html

**Comment by h-mayorquin** [comment]
> I thought this was a multi-file style of the format? So shouldn't there be a new interface for Intan that takes `folder_path` in that case?

Good question. The file always has a header `info.rhd` that contains the available streams and channels with standarized names. So the format always point ot the this `info` file.

File: CHANGELOG.md (hunk 1 above)
Comment (CodyCBakerPhD) [reply CHANGELOG.md:3]: Yes, deprecations first makes sense to me

File: src/neuroconv/datainterfaces/ecephys/requirements.txt (hunk 3 above)
Comment (h-mayorquin) [reply src/neuroconv/datainterfaces/ecephys/requirements.txt:2]: I don't think so.
But, once they merge this https://github.com/SpikeInterface/spikeinterface/pull/2833 or they release probe interface then the ceiling on the version can be removed.

**Comment by CodyCBakerPhD** [comment]
> Good question. The file always has a header info.rhd that contains the available streams and channels with standarized names. So the format always point ot the this info file.

Oh, I see

Sometimes the `.rhd` is a header + binary content

Other times it's just `.rhd` header and binaries live separate?

**Comment by CodyCBakerPhD** [comment]
Changes look fine and minimal to me

But what is your suggestion on release timing then? Since tests are passing, this all works with latest SI release, etc.? So we could cut a new release of NeuroConv without issue?

File: CHANGELOG.md (hunk 1 above)
Comment (h-mayorquin) [reply CHANGELOG.md:9]: Done.

**Comment by h-mayorquin** [comment]
Are you planning on making a release soon? I think that @alejoe91 can help us to merge https://github.com/SpikeInterface/spikeinterface/pull/2833 before the next release and then we can eliminate the ceiling.

**Comment by h-mayorquin** [comment]
> > Good question. The file always has a header info.rhd that contains the available streams and channels with standarized names. So the format always point ot the this info file.
> 
> Oh, I see
> 
> Sometimes the `.rhd` is a header + binary content
> 
> Other times it's just `.rhd` header and binaries live separate?

Yes.

**Comment by CodyCBakerPhD** [comment]
> Are you planning on making a release soon? I think that @alejoe91 can help us to merge https://github.com/SpikeInterface/spikeinterface/pull/2833 before the next release and then we can eliminate the ceiling.

Is the timeline on that in the next 2-3 weeks? It's been a while on the NeuroConv side and the GUIDE also releases in early June so would be nice to keep things synchronized

**Comment by h-mayorquin** [comment]
I think the confusion is this. 
* We don't need the latest SI release for this. This will work with the current bounds that you have in main.
* But the current bounds that you have in main will make the dev tests fail. 

I guess I will just eliminate the ceiling here but then you will have the dev tests failing.

**Comment by CodyCBakerPhD** [comment]
Thanks for clarifying; any ETA on the 'fix' that would otherwise happen on dev branch testing then?

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by h-mayorquin** [comment]
I am not privy to spikeinterface latest plans. I will ask Alessio next time we meet.

**Pull request merged**